File: 637/main.go
package main

import "fmt"

func main() {
	tr := &TreeNode{
		3,
		&TreeNode{9, nil, nil},
		&TreeNode{20,
			&TreeNode{15, nil, nil},
			&TreeNode{7, nil, nil},
		},
	}
	fmt.Println(averageOfLevels(tr))
}

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func averageOfLevels(root *TreeNode) []float64 {
	queue := []*TreeNode{root}
	result := []float64{}
	if root == nil {
		return result
	}
	for len(queue) != 0 {
		newQueue := []*TreeNode{}
		var sum float64
		for _, node := range queue {
			sum += float64(node.Val)
			if node.Left != nil {
				newQueue = append(newQueue, node.Left)
			}
			if node.Right != nil {
				newQueue = append(newQueue, node.Right)
			}
		}
		result = append(result, sum/float64(len(queue)))
		queue = newQueue
	}
	return result
}

File: 637/main.py
from typing import List, Optional


# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def averageOfLevels(self, root: Optional[TreeNode]) -> List[float]:
        """
        while level:
        result.append(avg)

        edge cases
        None
        """
        if root is None:
            return []
        result = []
        nodes = [root]
        while nodes:
            result.append(sum([n.val for n in nodes]) / len(nodes))
            new_nodes = []
            for node in nodes:
                if node.left:
                    new_nodes.append(node.left)
                if node.right:
                    new_nodes.append(node.right)
            nodes = new_nodes
        return result
